test/calibration/experiments/test_drag.py

```diff
@@ -96,6 +97,27 @@ def test_end_to_end(self):
         self.assertTrue(abs(result.value.n - backend.ideal_beta) < self.test_tol)
         self.assertEqual(result.quality, "good")
 
+    @data(
+        (0.0050, 1.0, 0.00, [1, 3, 5], None),  # partial oscillation.
+        (0.0025, 0.5, 0.00, [1, 3, 5], None),  # even slower oscillation with amp < 1
+        (0.0050, 0.8, 0.05, [3, 5, 7], None),  # constant offset, i.e. lower SNR.
+        (0.1000, 0.5, 0.10, [1, 3, 5], np.linspace(-1, 1, 51)),  # Beta not in range
```

Comment:
I like these cases but some of them are a little too nasty. The middle two in particular are problematic. In your plots, you can see that the error bars on beta are around 0.1 which is the tolerance for the tests. I tried running locally and changing `rng_seed` and saw that I got values outside of the tolerance. We shouldn't rely on the seed, so I think we should either relax the tolerance for those cases or adjust the parameters so there is less error in the fit.

Reply:
Agreed. I played around with the seed and adjusted the tol on the second one only. The third test seemed fine. [e11e1da](https://github.com/Qiskit/qiskit-experiments/pull/732/commits/e11e1da4ae89e4a3d1fe3545cdc042596d03984f)

Reply:
Actually, the third test is kind of right on the edge. The fit error is 0.05, so it will fall outside the tolerance about 4% of the time. We fix the seed so this doesn't matter too much, but I might bump the tolerance 0.2 just to make it really safe.